aioacaia/acaiascale.py

```diff
@@ -138,6 +142,45 @@ def timer(self) -> int:
             return 0
 
         return int(self._timer_stop - self._timer_start)
+    
+    @property
+    def flow_rate(self) -> float:
+
+        if len(self.weight_history) < 4:
+            return None
+
+        now = time.time()
+        flows = []
+
+        # Calculate flow rates using 3 readings ago
+        for i in range(3, len(self.weight_history)):
+            prev_time, prev_weight = self.weight_history[i - 3]
+            curr_time, curr_weight = self.weight_history[i]
+
+            time_diff = curr_time - prev_time
+            weight_diff = curr_weight - prev_weight
+
+            # Ignore weights older than 5 seconds
+            if now - prev_time > 5:
+                continue

```

Comment:
this I don't understand tbh. Shouldn't we just drop the items from the list then?

Reply:
I adjusted this to remove entries older than 5 seconds before the loop begins.